pkg/controller/util.go

```diff
@@ -1176,6 +1181,27 @@ func inflateSizeWithOverhead(c client.Client, imgSize int64, pvcSpec *v1.Persist
 	return returnSize, nil
 }
 
+// handleFailedPod handles pod-creation errors and updates the pod's PVC without providing sensitive information
+func handleFailedPod(err error, podName string, pvc *v1.PersistentVolumeClaim, recorder record.EventRecorder, c client.Client) error {
+	// Generic reason and msg to avoid providing sensitive information
```

Comment:
If you add `if err == nil { return nil }` you can rewrite all the ifs calling this func to:
```
if err := handleFailedPod(...); err != nil {
  return ...err...
}
```
which I think is nicer than calling it in the return call

Reply:
Yeah that's a good one, I'll change it :+1: